readthedocs/vcs_support/base.py

```diff
@@ -101,6 +101,11 @@ def run(self, *cmd, **kwargs):
 
         try:
             build_cmd = self.environment.run(*cmd, **kwargs)
+        except BuildCancelled as e:
+            # Catch ``BuildCancelled`` here and re raise it. Otherwise, if we
+            # raise a ``RepositoryError`` then the ``on_failure`` method from
+            # Celery won't treat this problem as a ``BuildCancelled`` issue.
+            raise BuildCancelled
```

Comment:
Not sure if there is any context in `e`, but perhaps this should be `raise` or `raise(BuildCancelled(e))`? This is currently tossing out the context. If the exception context isn't required, `as e` above can be dropped